CMakeLists.txt

```diff
@@ -311,10 +311,125 @@ add_subdirectory(pkgconfig)
 # to define the test executables.
 #
 if(ENABLE_TESTING OR ENABLE_PROGRAMS)
+    add_custom_command(
+        OUTPUT
+            ./tests/src/test_keys.h
+        WORKING_DIRECTORY
+            ${CMAKE_CURRENT_SOURCE_DIR}
+        COMMAND
+            ${MBEDTLS_PYTHON_EXECUTABLE}
+                "${CMAKE_CURRENT_SOURCE_DIR}/tests/scripts/generate_test_keys.py"
+                    "--output"
+                        "${CMAKE_CURRENT_SOURCE_DIR}/tests/src/test_keys.h"
+        DEPENDS
+            ${CMAKE_CURRENT_SOURCE_DIR}/tests/scripts/generate_test_keys.py
+    )
+    add_custom_command(
+        OUTPUT
+            ./tests/src/test_certs.h
+        WORKING_DIRECTORY
+            ${CMAKE_CURRENT_SOURCE_DIR}
+        COMMAND
+            "${MBEDTLS_PYTHON_EXECUTABLE}"
+                "${CMAKE_CURRENT_SOURCE_DIR}/tests/scripts/generate_test_cert_macros.py"
```

Comment:
That's a very long list of parameters that needs to be repeated in many places, so I'm requesting a design change:

* `generate_test_cert_macros.py` with no arguments just works. All the file names involved and other data are stored in the script.
* `generate_test_cert_macros.py --list-dependencies` (option name TBD, I don't think we have an established standard here) lists the dependencies of the script. Call this in `Makefile` and `CMakeLists.txt`.